Update README.md for New Era (#112)
JuliaDiff/ChainRules.jl: README.md

[![Travis](https://travis-ci.org/JuliaDiff/ChainRules.jl.svg?branch=master)](https://travis-ci.org/JuliaDiff/ChainRules.jl)
[![Coveralls](https://coveralls.io/repos/github/JuliaDiff/ChainRules.jl/badge.svg?branch=master)](https://coveralls.io/github/JuliaDiff/ChainRules.jl?branch=master)

**Docs:**
[![](https://img.shields.io/badge/docs-master-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/dev)
[![](https://img.shields.io/badge/docs-stable-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/stable)

The ChainRules package provides a variety of common utilities that can be used by downstream automatic differentiation (AD) tools to define and execute forward-, reverse-, and mixed-mode primitives.

The Core logic of ChainRules is implemented in [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl).

> If adding ChainRules support to your package (defining `rrule`s or `ffules` you only need to depend on that very light-weight package.

Reviewer: ```suggestion
To add ChainRules support to your package, by defining new `rrule`s or `ffrules`, you only need to depend on the very light-weight package ChainRulesCore.jl.
```

> This repo: ChainRules.jl is what people actually use, it reexports all the ChainRules functionality,

Reviewer: can we make these full sentences? e.g.
```suggestion
This repository contains ChainRules.jl, which is what people actually use directly. 
ChainRules reexports all the ChainRulesCore functionality, and has all the rules for the Julia standard library.
```
and one sentence per line makes diffs in follow-up PRs much nicer

Reviewer: here @oxinabox

> and has all the rules for the julia standard library.

Reviewer: ```suggestion
```

Author: we have to say something like that.

Reviewer: this is to change to 1 sentence per line.

the content is included in the new full sentence in the suggestion one up (at https://github.com/JuliaDiff/ChainRules.jl/pull/112#discussion_r328636209)

Here are some of the core features of the package:

- Mixed-mode composability without being coupled to a specific AD implementation.

> - Extensible rules: package authors can add rules (and thus AD support) to the functions in their packages in their packages. They don't need to edit the ChainRules repo.

Reviewer: ```suggestion
- Extensible rules: package authors can add rules (and thus AD support) to the functions in their packages, without needing to make a PR to ChainRules.jl .
```

Author: idk

Reviewer: at least change it to remove the repeated words?

> - Control-inverted design: rule authors can fully specify derivatives in a concise manner while naturally allowing the caller to compute only what they need.

Reviewer: what does this mean?

Reviewer: actually i think i know what it means, but it's taken me a couple readings to get there...

How about changing the last bit to something like this (still not totally happy with it)

```suggestion
- Control-inverted design: rule authors can fully specify derivatives in a concise manner that supports computation efficiencies, so we will only compute as much as the user requests.
```

[![Travis](https://travis-ci.org/JuliaDiff/ChainRules.jl.svg?branch=master)](https://travis-ci.org/JuliaDiff/ChainRules.jl)
[![Coveralls](https://coveralls.io/repos/github/JuliaDiff/ChainRules.jl/badge.svg?branch=master)](https://coveralls.io/github/JuliaDiff/ChainRules.jl?branch=master)

**Docs:**
[![](https://img.shields.io/badge/docs-master-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/dev)
[![](https://img.shields.io/badge/docs-stable-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/stable)

The ChainRules package provides a variety of common utilities that can be used by downstream automatic differentiation (AD) tools to define and execute forward-, reverse-, and mixed-mode primitives.

> The Core logic of ChainRules is implemented in [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl).

Author: ```suggestion
The core logic of ChainRules is implemented in [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl).
```

> To add ChainRules support to your package, by defining new `rrule`s or `ffrules`, you only need to depend on the very light-weight package ChainRulesCore.jl.

Author: ```suggestion
To add ChainRules support to your package, by defining new `rrule`s or `frules`, you only need to depend on the very light-weight package ChainRulesCore.jl.
```
